test/net-api/test-connection.js

```diff
@@ -0,0 +1,107 @@
+var tape = require('tape')
+var utp = require('../../index.js')
+
+tape('net-api connection', function (t) {
+  t.test('connect', function (t) {
+    var connected = false
+
+    utp.createServer(function () {
+      connected = true
+    }).listen(53410)
+
+    var socket = utp.connect(53410)
+
+    socket.on('connect', function () {
+      t.ok(connected, 'connected successfully')
+      t.end()
+    })
+  })
+
+  t.test('connect deferred', function (t) {
+    var connected = false
+
+    setTimeout(function () {
+      utp.createServer(function () {
+        connected = true
+      }).listen(53400)
+    }, 100)
+
+    var socket = utp.connect(53400)
+    socket.on('connect', function () {
+      t.ok(connected, 'connected successfully')
+      t.end()
+    })
+  })
+
+  t.skip('close', function (t) {
+    var server = utp.createServer(function (socket) {
+      socket.resume()
+      socket.end()
+      // TODO callback on server.close is not implemented
+      server.close(function () {
+        t.pass('closed')
+        t.end()
+      })
```

Comment:
would you prefer to pass the callback around (https://github.com/mafintosh/utp-native/blob/master/index.js#L139-L141) or just add it as a one time listener for the `close` event that eventually gets emitted?